src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
+
+@ApiTags('Auth')
+@Controller()
+export class AuthController {
+  constructor(private readonly authService: AuthService) {}
+
+  @Get('oauth/callback')
+  async githubAuthCallback(@Query('code') code: string) {
+    try {
+      const jwtToken = await this.authService.handleGitHubCallback(code);
+
+      return jwtToken;
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @ApiBearerAuth()
+  @UseGuards(JwtAuthGuard)
+  @Post('logout')
+  async logout(@Req() req: TRequest) {
+    try {
+      return await this.authService.logout(req);
+    } catch (error) {
+      if (error.status)
+        throw new HttpException(error.message, error.status);
+      else throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+    }
```

Comment:
PullProofAI

Ensure consistent error handling in the `logout` method by following a similar pattern as in the `githubAuthCallback` method.

```diff
+       if (error instanceof CustomError) {
+         // Handle custom errors with specific status codes
+         throw new HttpException(error.message, error.status);
+       } else {
+         // Handle generic errors with a default status code
+         throw new HttpException(error.message, HttpStatus.INTERNAL_SERVER_ERROR);
+       }
```

#### Overall Comments
- The controller methods seem well-structured and follow NestJS conventions.
- Consider extracting common error handling logic into a reusable function to avoid duplication and improve maintainability.
- Ensure consistency in error handling across all controller methods for a better user experience and easier debugging.


<!-- This is an auto-generated comment by PullProofAI -->